src/node/node_state.h

```diff
@@ -1188,6 +1188,14 @@ namespace ccf
       tx2->set_flag(kv::AbstractStore::Flag::LEDGER_CHUNK_AT_NEXT_SIGNATURE);
     }
 
+    void request_snapshot(kv::Tx& tx) override
+    {
+      snapshotter->forced_snapshot_after_idx =
+        network.tables->current_version();
+      snapshotter->set_flag_unsafe(
+        ccf::Snapshotter::Flag::FORCE_SNAPSHOT_AT_NEXT_SIGNATURE);
```

Comment:
This is a side effect that is not tied to a specific `seqno` so it feels unsafe (how is it rolled back, etc.?). I would expect this to set a flag on `tx` which is later passed on the `store` and the store passes it to the snapshotter (through `raft` or better, #3026).

Reply:
It is tied to the tx index via `forced_snapshot_after_idx`, which can be rolled back there if we want to keep that variable. I didn't see a way for the store to pass this on to the snapshotter (it doesn't have a reference to the snapshotter or raft, but that may change after #3026 is done).